src/topology/mod.rs

```diff
@@ -759,41 +804,49 @@ impl TopologyDescription {
     }
 
     // Begins monitoring hosts that are not currently being monitored.
-    fn add_missing_hosts(&mut self,
-                         description: &ServerDescription,
-                         client: Client,
-                         top_arc: Arc<RwLock<TopologyDescription>>,
-                         run_monitor: bool) {
-
-        for host in &description.hosts {
+    fn add_missing_hosts(
+        &mut self,
+        description: Arc<RwLock<ServerDescription>>,
+        client: Client,
+        top_arc: Arc<RwLock<TopologyDescription>>,
+        run_monitor: bool,
+    ) {
+
+        for host in &description.read().unwrap().hosts {
             if !self.servers.contains_key(host) {
-                let server = Server::new(client.clone(),
-                                         host.clone(),
-                                         top_arc.clone(),
-                                         run_monitor,
-                                         self.stream_connector.clone());
+                let server = Server::new(
+                    client.clone(),
+                    host.clone(),
+                    top_arc.clone(),
+                    run_monitor,
+                    self.stream_connector.clone(),
+                );
                 self.servers.insert(host.clone(), server);
             }
         }
 
-        for host in &description.passives {
+        for host in &description.read().unwrap().passives {
             if !self.servers.contains_key(host) {
-                let server = Server::new(client.clone(),
-                                         host.clone(),
-                                         top_arc.clone(),
-                                         run_monitor,
-                                         self.stream_connector.clone());
+                let server = Server::new(
+                    client.clone(),
+                    host.clone(),
+                    top_arc.clone(),
+                    run_monitor,
+                    self.stream_connector.clone(),
+                );
                 self.servers.insert(host.clone(), server);
             }
         }
 
-        for host in &description.arbiters {
+        for host in &description.read().unwrap().arbiters {
```

Comment:
same here — wondering if we should clone these vecs instead of locking the description?